src/Sentry.Maui/BreadcrumbEvent.cs

```diff
@@ -0,0 +1,50 @@
+namespace Sentry.Maui;
+
+/// <summary>
+/// Argument to the OnBreadcrumbCreateCallback
+/// </summary>
+public sealed class BreadcrumbEvent
+{
+    /// <summary>
+    /// The sender of the event, usually the control that triggered it.
+    /// </summary>
+    public object? Sender { get; }
+
+    /// <summary>
+    /// The event name (e.g. "Tapped", "Swiped", etc.)
+    /// </summary>
+    public string EventName { get; }
+
+    /// <summary>
+    /// Any extra data to be included in the breadcrumb. This would typically be event specific information (for example
+    /// it could include the X, Y coordinates of a tap event).
+    /// </summary>
+    public (string Key, string Value)[] ExtraData { get; }
```

Comment:
why did we end up with a tuple array?
Is this preferred these days over IList or ICollection or something?
We don't want to allow one to grow it (no Add?)

Reply:
Originally `BreadcrumbEvent` was a record (not a class)... so it's not supposed to be mutable. All of the properties of this class are getter only can only be set via the constructor.

It's basically used to communicate the details to create a breadcrumb from the UI events (like tap or drag or whatever) to a callback like this:
https://github.com/getsentry/sentry-dotnet/blob/94ee8f6faebae3849b293dace84bd3213daeea4c/src/Sentry.Maui/Internal/MauiEventsBinder.cs#L128-L144

If we don't like the Tuple array, other options would be `KeyValuePair[]` or `Span<KeyValuePair>` maybe?

Reply:
Ah, I see what you're saying... we don't need a specific concrete type here... we could leave it flexible (like `ICollection`).

I can do that 👍🏻

Reply:
OK, changed to:
https://github.com/getsentry/sentry-dotnet/blob/dbf9985f8e1afe1e636bb9673f54b8578723950b/src/Sentry.Maui/BreadcrumbEvent.cs#L22

I added a constructor overload so that people can pass tuples or KeyValuePairs but ultimately stored internally like this.

Reply:
Good catch. We intended this type to be immutable. But a `System.Array` is not immutable.